### PR TITLE
Refactor: Improve Code Readability with Constants and Early Returns

### PR DESCRIPTION
**Pull request by k-vanio**
Refactor: Improve Code Readability with Constants and Early Returns

This commit refactors the `realIP` function by introducing constants for header names
(`trueClientIP`, `xRealIP`, `xForwardedFor`) to enhance code clarity. Additionally, it
utilizes the concept of early returns for simplifying the logic and improving overall
readability.

These changes contribute to better maintainability and understanding of the codebase.

### CODE REVIEW

**Comment by lesichkovm** [comment]
@k-vanio, you have done good job, two points:

1. The constants should be alphabetically sorted to improve readability
2. You have missed the net.ParseIP(ip) logic which was checking the IP for validity

**Comment by k-vanio** [comment]
@lesichkovm Thank you for your feedback! I've addressed the first point by alphabetically sorting the constants. However, regarding the second point about the `net.ParseIP(ip)` logic, I couldn't identify any issues with it as it seems correct to me. Could you please provide more details or context about what you've observed?

**Comment by lesichkovm** [comment]
Did not see the new function **isValidIP**. All is good @k-vanio

File: middleware/realip.go [hunk 1]
```diff
@@ -40,21 +42,25 @@ func RealIP(h http.Handler) http.Handler {
...
-	} else if xrip := r.Header.Get(xRealIP); xrip != "" {
-		ip = xrip
-	} else if xff := r.Header.Get(xForwardedFor); xff != "" {
+	if xff := r.Header.Get(xForwardedFor); xff != "" {
 		i := strings.Index(xff, ",")
 		if i == -1 {
 			i = len(xff)
 		}
-		ip = xff[:i]
+		return xff[:i]
```
Comment (VojtechVitek) [inline middleware/realip.go:58]: looks like you're not checking if XFF is valid IP

**Review by VojtechVitek — CHANGES_REQUESTED** [review]

File: middleware/realip.go (hunk 1 above)
Comment (neio-vanio) [reply middleware/realip.go:58]: I greatly appreciate you sharing your observation. I will review the code and implement the necessary checks to ensure that the IP address present in the X-Forwarded-For (XFF) header is validated correctly. Thank you for bringing this to my attention!

Comment (k-vanio) [reply middleware/realip.go:58]: I greatly appreciate you sharing your observation. I will review the code and implement the necessary checks to ensure that the IP address present in the X-Forwarded-For (XFF) header is validated correctly. Thank you for bringing this to my attention!

File: middleware/realip.go [hunk 2]
```diff
@@ -40,21 +42,28 @@ func RealIP(h http.Handler) http.Handler {
...
 		if i == -1 {
 			i = len(xff)
 		}
-		ip = xff[:i]
-	}
-	if ip == "" || net.ParseIP(ip) == nil {
-		return ""
+
+		if isValidIP(xff[:i]) {
+			return xff[:i]
```
Comment (VojtechVitek) [inline middleware/realip.go:60]: this is slicing twice unnecessary

Comment (VojtechVitek) [inline middleware/realip.go:60]:
```
if i := strings.Index(xff, ","); i >= 0 {
  xff = xff[:i]
}
```

Comment (VojtechVitek) [reply middleware/realip.go:60]:
```
if i := strings.Index(xff, ","); i >= 0 {
  xff = xff[:i]
}
```

Comment (k-vanio) [reply middleware/realip.go]: Thank you! Your contribution to refining XFF's IP address validation is very good.

**Review by VojtechVitek — APPROVED** [review]
LGTM

**Comment by PauloRobertTlss** [comment]
LGTM 🚀

**Review by VojtechVitek — COMMENTED** [review]
Hi @k-vanio,

Thanks for the Pull Request and sorry it did't get merged to the `master` branch. 

I decided to merge the following PR instead, which adds a new feature and also improves modularity and readability at the same time:

Superseded by https://github.com/go-chi/chi/pull/908.

Best,
Vojtech

**Pull request closed without merging**